src/Storage/Factory/FileSystemFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace EMS\CommonBundle\Storage\Factory;
+
+use EMS\CommonBundle\Storage\Service\FileSystemStorage;
+use EMS\CommonBundle\Storage\Service\StorageInterface;
+use Psr\Log\LoggerInterface;
+
+class FileSystemFactory implements StorageFactoryInterface
+{
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /** @var string[]  */
+    private $usedFolder = [];
+
+    /**
+     * @param array<mixed> $parameters
+     * @return StorageInterface
+     */
+    public function createService(array $parameters): ?StorageInterface
+    {
+        if ('fs' !== $parameters['type'] ?? null) {
+            throw new \RuntimeException('The storage service type doesn\'t match \'fs\'');
+        }
```

Comment:
This is repeated in each file. By using a protected CONST STORAGE_TYPE this can be defined in a AbstractStorageFactory class?

Reply:
I'm now using the Symfony OptionResolver